ls1intum/Artemis
# `Notifications`: Do not send a notification when a user was removed from a conversation in case this notification is disabled

### Checklist
#### General



### Motivation and Context


Even though users disable the conversation notification from settings when they reload they still get notification when they were removed from a conversation.

### Description

In class `NotificationSettingsService` i added  `CONVERSATION_REMOVE_USER_GROUP_CHAT` and `CONVERSATION_REMOVE_USER_CHANNEL` to `NOTIFICATION_SETTING_ID_TO_NOTIFICATION_TYPES_MAP` map. 

### Steps for Testing

Prerequisites:
- 2 users registered at the same course
- 1 Course

1. Log in to Artemis
2. Navigate to notification settings as user2 and disable notifications about conversations (`New messages in conversations`)
3. Now on other tab logged in as user1 navigate to messages tab and start chat with the other user
4. Create a group chat or a channel then add the second user to the group chat or channel. 
5. User2 should not receive notifications.
6. As user1 now remove user2 from channel or group chat.
7. User2 should not receive notification about removal and also **reload** the page, still user2 should not get notifications about removal on sidebar.

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test
